From 5b18e11e27e59c6e72c69f63e01427e86bc24b99 Mon Sep 17 00:00:00 2001
From: Zidan Wang <zidan.wang@freescale.com>
Date: Thu, 7 May 2015 13:58:27 +0800
Subject: [PATCH 0531/1221] MLK-10843-1 ASoC: imx-wm8960: set codec mclk in
 device tree instead of machine driver

set codec mclk in device tree instead of machine driver.

Signed-off-by: Zidan Wang <zidan.wang@freescale.com>
[zou:Original patch taken from
git.freescale.com/imx/fsl-arm-yocto-bsp.git-b imx-3.14.52-1.1.0_ga]
Signed-off-by: zou cao <cao.zou@windriver.com>
---
 sound/soc/fsl/imx-wm8960.c |    8 --------
 1 files changed, 0 insertions(+), 8 deletions(-)

diff --git a/sound/soc/fsl/imx-wm8960.c b/sound/soc/fsl/imx-wm8960.c
index eb528b9..46249fe 100644
--- a/sound/soc/fsl/imx-wm8960.c
+++ b/sound/soc/fsl/imx-wm8960.c
@@ -27,7 +27,6 @@
 #include "fsl_sai.h"
 
 #define DAI_NAME_SIZE	32
-#define DEFAULT_MCLK_FREQ (12000000)
 
 struct imx_wm8960_data {
 	struct snd_soc_card card;
@@ -313,13 +312,6 @@ static int imx_hifi_hw_params(struct snd_pcm_substream *substream,
 	}
 
 	data->clk_frequency = clk_get_rate(data->codec_clk);
-	if (data->clk_frequency != DEFAULT_MCLK_FREQ) {
-		ret = clk_set_rate(data->codec_clk, DEFAULT_MCLK_FREQ);
-		if (ret < 0)
-			dev_dbg(dev, "fail to set codec clock frequency.\n");
-		else
-			data->clk_frequency = clk_get_rate(data->codec_clk);
-	}
 
 	/* Use MCLK to provide sysclk directly*/
 	sysclk = data->clk_frequency;
-- 
1.7.5.4

